pydantic/generics.py

```diff
@@ -62,7 +62,13 @@ def __class_getitem__(cls: Type[GenericModelT], params: Union[Type[Any], Tuple[T
             returned as is.
 
         """
-        cached = _generic_types_cache.get((cls, params))
+
+        def _cache_key(
+            _params: Union[Type[Any], Tuple[Type[Any], ...]]
+        ) -> Tuple[Type[GenericModelT], Union[Type[Any], Tuple[Type[Any], ...]], Tuple[Any, ...]]:
```

Comment:
Just use, this is an internal method, so that's sufficient.

```suggestion
        def _cache_key(_params: Any) -> Tuple[Type[GenericModelT], Any, Tuple[Any, ...]]:
```

Reply:
Ah! Tried to replace with `Type[Any]` instead of `Any` but that did not work.

Also, I was wondering about closing the issue, as the nested variant of the issue is not fixed. Do you want to keep the issue open for this or is the documented and skipped test good enough?

Reply:
I don't mind, if you want to keep the issue open, use the PR ID in the change file.

Either way this is urgent since people are (understandably) waiting on the patch release.

I just want this to pass so I can merge it and get the release done.

Reply:
Added 'fix'. You can reopen the issue after, if needed.

Reply:
great, thanks.